includes/razorpay-subscriptions.php

```diff
@@ -124,10 +132,86 @@ protected function getSubscriptionCreateData($orderId)
         return $subscriptionData;
     }
 
-    protected function getProductPlanId($product, $order)
+    /**
+     * @param $subscriptionData
+     * @param $signUpFee
+     * @param $order
+     * @throws Errors\Error
+     */
+    protected function setStartAtIfNeeded(& $subscriptionData, & $signUpFee, $order)
     {
-        $currency = get_woocommerce_currency();
+        $product = $this->getProductFromOrder($order);
+
+        $sub = $this->getWooCommerceSubscriptionFromOrderId($order->get_id());
+
+        $metadata = get_post_meta($product['product_id']);
+
+        if (empty($metadata['razorpay_wc_start_date']) === false)
+        {
+            //
+            // When custom start date is set, we consider the initial payment
+            // as a sign up payment, and the first recurring payment will be
+            // made on the configured start date on the next month
+            //
+            $startDay = $metadata['razorpay_wc_start_date'][0];
+
+            $startDay = 40;
+
+            //
+            // $startDay must be in between 1 and 28
+            //
+            if (($startDay <= 1) or ($startDay >= 28))
+            {
+                throw new Errors\Error(
+                    'Invalid start day saved as subscription product metadata',
+                    WooErrors\SubscriptionErrorCode::SUBSCRIPTION_START_DATE_INVALID,
+                    400
+                );
+            }
+
+            $startDate = $this->getStartDate($startDay, $sub);
+
+            // We modify the sign up fee which was passed by reference
+            $signUpFee += $sub->get_total();
+
+            $subscriptionData['start_at'] = $startDate;
+
+            //
+            // In the case where we take the first recurring payment as a up front amount, and the
+            // second recurring payment as the first recurring payment, we reduce the total count by 1
+            //
+            $subscriptionData['total_count'] = $subscriptionData['total_count'] - 1;
+        }
+    }
 
+    protected function getStartDate($startDay, $sub)
```

Comment:
Most merchants want to keep the start_at to remain null so that the immediate charge includes the first subscription amount. `start_at` only needs to be set IFF the start_date is set in metadata or if a trial period is set.

Reply:
@captn3m0 Already implemented in `setStartAtAndSignUpFeeIfNeeded`